flow/layers/display_list_layer_unittests.cc

```diff
@@ -294,6 +294,56 @@ TEST_F(DisplayListLayerTest, CachedIncompatibleDisplayListOpacityInheritance) {
   EXPECT_TRUE(DisplayListsEQ_Verbose(expected.Build(), this->display_list()));
 }
 
+TEST_F(DisplayListLayerTest, RasterCachePreservesRTree) {
+  const SkRect picture1_bounds = SkRect::MakeXYWH(10, 10, 10, 10);
+  const SkRect picture2_bounds = SkRect::MakeXYWH(15, 15, 10, 10);
+  DisplayListBuilder builder(true);
+  builder.DrawRect(picture1_bounds, DlPaint());
+  builder.DrawRect(picture2_bounds, DlPaint());
+  auto display_list = builder.Build();
+  auto display_list_layer = std::make_shared<DisplayListLayer>(
+      SkPoint::Make(3, 3), display_list, true, false);
+
+  use_skia_raster_cache();
+
+  auto context = preroll_context();
+  display_list_layer->Preroll(preroll_context());
+  EXPECT_EQ(context->renderable_state_flags, 0);
+
+  // Pump the DisplayListLayer until it is ready to cache its DL
+  display_list_layer->Preroll(preroll_context());
+  display_list_layer->Preroll(preroll_context());
+  display_list_layer->Preroll(preroll_context());
+  LayerTree::TryToRasterCache(*preroll_context()->raster_cached_entries,
+                              &paint_context(), false);
+
+  DisplayListBuilder expected_root_canvas(true);
+  context->raster_cache->Draw(display_list_layer->caching_key_id(),
```

Comment:
Doesn't this return a boolean that it succeeded? You should ASSERT_EQ on that, otherwise the rest of the test is invalid.

Reply:
Same below.